docs/CToSwiftNameTranslation.md

```diff
@@ -190,4 +195,48 @@ extension SecretResourceID {
+
+
+## CF Types
+
+"Core Foundation" is a C-based object-oriented system with strong conventions built around pointers to opaque structs. Creating new Core Foundation types is not generally supported, but Swift will recognize those in Apple's SDKs. If a struct has the `objc_bridge`, `objc_mutable_bridge`, or `objc_bridge_related` Clang attributes, it will be treated as a CF type and a typedef of a pointer to that struct will be imported as a class in Swift. The suffix "Ref" will be dropped from the class's name if present unless doing so would conflict with another declaration in the same module as the typedef.
+
+If the class name contains the word "Mutable" exactly once per the usual word-boundary rules, a corresponding class name without the word "Mutable" will be used as the superclass if present. Otherwise, the CF type is taken to be a root object.
+
+Additionally, typedefs for `void *` or `const void *` that are themselves annotated with `objc_bridge` will be treated as CFTypeRef-like and imported as `Any` rather than `Unsafe[Mutable]RawPointer`.
+
+If a typedef's underlying type is itself a "CF pointer" typedef, the "alias" typedef will be imported as a regular typealias, with the suffix "Ref" still dropped from its name (if present) unless doing so would conflict with another declaration in the same module as the typedef.
+
+
+## `swift_private`
+
+The `swift_private` Clang attribute prepends `__` onto the base name of any declaration being imported except initializers. For initializers with no arguments, a dummy `Void` argument with the name `__` is inserted; otherwise, the label for the first argument has `__` prepended. This transformation takes place after any other name manipulation, unless the declaration has a custom name. It will not occur if the declaration is an override; in that case the name needs to match the overridden declaration.
+
+```objc
+@interface Example : NSObject
+- (instancetype)initWithValue:(int)value __attribute__((swift_private));
+@property(readonly) int value __attribute__((swift_private));
+@end
+
+// Usually seen as NS_REFINED_FOR_SWIFT
+```
+
+```swift
+class Example: NSObject {
+  init(__value: Int32)
+  var __value: Int32 { get }
+}
+```
+
+The purpose of this annotation is to allow a more idiomatic implementation to be provided in Swift. The effect of `swift_private` is inherited from an enum onto its elements if the enum is not imported as an error code enum, an `@objc` enum, or an option set.
+
+_The original intent of the `swift_private` attribute was additionally to limit access to a Swift module with the same name as the owning Clang module, e.g. the Swift half of a mixed-source framework. However, this restriction has not been implemented as of Swift 5.1._
+
+_For "historical reasons", the `swift_private` attribute is ignored on factory initializers with no arguments._
```

Comment:
I traced this back to 91d1c3b21a, and it looks like there's no *motivation* for this; it was just "we forgot to implement it in the factory-method-to-initializer transformation and then didn't want to break anyone who wrote `swift_private` on a factory method and saw no effect". The code lives in ImportName.cpp today; that's where I got the "historical reasons" phrasing from.

Reply:
Your phrasing uses "factory initializers", is it the same as "factory methods"?

Reply:
Mm, not quite, because only some factory methods are imported as initializers. And that section hasn't been written yet. I can expand it out to "factory methods with no arguments that are imported as initializers", though.